src/screen.rs

```diff
@@ -669,7 +669,20 @@ impl Screen {
         const CTRL: Modifiers = Modifiers::CTRL;
         const NONE: Modifiers = Modifiers::NONE;
         const SHIFT: Modifiers = Modifiers::SHIFT;
-        match (key.modifiers, key.key) {
+
+        // Normalize characters on US keyboard. This is useful on Windows. Note
+        // termwiz 0.7 only normalizes ASCII characters (Shift + 'C' => 'C') and
+        // does not assume US keyboard.
+        let modifiers = if cfg!(windows)
+            && key.modifiers.contains(SHIFT)
+            && ":{}()<>#?".chars().any(|ch| key.key == Char(ch))
```

Comment:
Ah, this stuff is tricky because it technically should know things about the keymap.  But, if we consider only SHIFT, I believe this statement is true regardless of the keymap, and the platform:

> If the character is ascii alphanumeric or ascii punctuation then the effect of the shift key has already been applied to produce the current character code.

If that is true then we can always just discard the SHIFT modifier from the effective set.

WIth that in mind, I think this code turns into this (which is a superset of the characters you were already testing); we only need to check for punctuation here because `normalize_shift_to_upper_case` already handled the alpha cases in the core termwiz code:

```
let modifiers = match key.key {
    Char(ch) if ch.is_ascii_punctuation() => key.modifiers - SHIFT,
    _ => key.modifiers
};
```

```

Reply:
Nice to know it's `is_ascii_punctuation` normalized in termwiz! I tested it using termwiz master and it works as expected. However termwiz 0.7.1 still requires manual normalization for some reason.

I bisected it to https://github.com/wez/wezterm/commit/fe890827647e96d480149949cfa5740c7ad9bea4. Maybe it's time for a new point release? Then this PR can then be just upgrading termwiz. (a funny note: I just learned that git bisect refuses to work for bad -> good transition. I now appreciate hg bisect more).

Reply:
I've published termwiz 0.7.2:

```
> cargo run --example key_tester
   Compiling termwiz v0.7.1 (C:\Users\wez\wez-personal\wezterm\termwiz)
    Finished dev [unoptimized + debuginfo] target(s) in 2.80s
     Running `C:\Users\wez\wez-personal\wezterm\target\debug\examples\key_tester.exe`
Key(KeyEvent { key: Escape, modifiers: NONE })
Key(KeyEvent { key: Char('\''), modifiers: NONE })
Key(KeyEvent { key: Shift, modifiers: SHIFT })
Key(KeyEvent { key: Char('{'), modifiers: NONE })
Key(KeyEvent { key: Char('['), modifiers: NONE })
Key(KeyEvent { key: Char('C'), modifiers: CTRL })
```